packages/lib/src/components/internal/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,78 @@
+import { h } from 'preact';
+import { useCallback, useMemo } from 'preact/hooks';
+import cx from 'classnames';
+import uuid from '../../../utils/uuid';
+import './Toggle.scss';
+
+interface ToggleProps {
+    label?: string;
+    labelPosition?: 'before' | 'after';
+    ariaLabel?: string;
+    description?: string;
+    checked: boolean;
+    disabled?: boolean;
+    readonly?: boolean;
+    onChange?(checked: boolean): void;
+}
+
+const Toggle = ({ label, labelPosition = 'after', ariaLabel, description, checked, disabled = false, readonly = false, onChange }: ToggleProps) => {
+    const descriptionId = useMemo(() => (description ? `toggle-description-${uuid()}` : null), [description]);
+    const computedAriaLabel = useMemo(() => ariaLabel ?? label, [ariaLabel, label]);
+
+    const conditionalClasses = cx({
+        'adyen-checkout-toggle--label-first': labelPosition === 'before',
+        'adyen-checkout-toggle--disabled': disabled,
+        'adyen-checkout-toggle--readonly': readonly
+    });
+
+    const onInputChange = useCallback(
+        (event: Event) => {
+            onChange((event.target as HTMLInputElement).checked);
+        },
+        [onChange]
+    );
+
+    return (
+        <label className={`adyen-checkout-toggle ${conditionalClasses}`}>
+            <input
+                disabled={disabled}
+                checked={checked}
+                onChange={onInputChange}
+                aria-label={computedAriaLabel}
+                aria-readonly={readonly}
+                aria-describedby={descriptionId}
+                role="switch"
+                type="checkbox"
+                className="adyen-checkout-toggle__input"
+            />
+
+            <span aria-hidden={true} className="adyen-checkout-toggle__track">
+                <span className="adyen-checkout-toggle__handle">
+                    {checked && (
+                        <svg role="img" xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="none">
```

Comment:
I think this part could potentially be refactored into a checkmark component, optionally though.

Reply:
Yeah I was thinking the same .. To be honest I wasn't sure what to do, since:
- we don't have SVG as components in our codebase (although I don't see a problem to have few bento SVG's as components, for example this checkmark icon)
- we have this checkmark on the CDN  - although I didn't want to add complexity to this component to use the `getImage` hook to download the checkmark. Feels like too much to a simple UI component. Doing this would add more complexity, external dependency, and complicate more unit tests